Analysis/EventFiltering/nucleiFilter.cxx

```diff
@@ -0,0 +1,139 @@
+  constants::physics::MassDeuteron, constants::physics::MassTriton,
+  constants::physics::MassHelium3, constants::physics::MassAlpha};
+static constexpr std::array<int, nNuclei> charges{1, 1, 2, 2};
+static const std::vector<std::string> nucleiNames{"H2", "H3", "He3", "He4"};
+static const std::vector<std::string> cutsNames{
+  "TPCnSigmaMin", "TPCnSigmaMax", "TOFnSigmaMin", "TOFnSigmaMax", "TOFpidStartPt"};
+static constexpr float cutsPID[nNuclei][nCutsPID]{
+  {-3.f, +3.f, -4.f, +4.f, 1.0f},    /*H2*/
+  {-3.f, +3.f, -4.f, +4.f, 1.6f},    /*H3*/
+  {-5.f, +5.f, -4.f, +4.f, 14000.f}, /*He3*/
+  {-5.f, +5.f, -4.f, +4.f, 14000.f}  /*He4*/
+};
+} // namespace
+
+struct nucleiFilter {
+
+  Produces<aod::NucleiFilters> tags;
+
+  Configurable<float> yMin{"yMin", -0.8, "Maximum rapidity"};
+  Configurable<float> yMax{"yMax", 0.8, "Minimum rapidity"};
+  Configurable<float> yBeam{"yBeam", 0., "Beam rapidity"};
+
+  Configurable<float> cfgCutVertex{"cfgCutVertex", 10.0f, "Accepted z-vertex range"};
+  Configurable<float> cfgCutEta{"cfgCutEta", 0.8f, "Eta range for tracks"};
+
+  Configurable<LabeledArray<float>> cfgCutsPID{"nucleiCutsPID", {cutsPID[0], nNuclei, nCutsPID, nucleiNames, cutsNames}, "Nuclei PID selections"};
+
+  HistogramRegistry spectra{"spectra", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+
+  void init(o2::framework::InitContext&)
+  {
+    std::vector<double> ptBinning = {0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.8, 2.0, 2.2, 2.4, 2.8, 3.2, 3.6, 4., 5.};
+    std::vector<double> centBinning = {0., 1., 5., 10., 20., 30., 40., 50., 70., 100.};
+
+    AxisSpec ptAxis = {ptBinning, "#it{p}_{T} (GeV/#it{c})"};
+    AxisSpec centAxis = {centBinning, "V0M (%)"};
+
+    spectra.add("fCollZpos", "collision z position", HistType::kTH1F, {{600, -20., +20., "z position (cm)"}});
+    spectra.add("fTPCsignal", "Specific energy loss", HistType::kTH2F, {{600, 0., 3, "#it{p} (GeV/#it{c})"}, {1400, 0, 1400, "d#it{E} / d#it{X} (a. u.)"}});
+    spectra.add("fTPCcounts", "n-sigma TPC", HistType::kTH2F, {ptAxis, {200, -100., +100., "n#sigma_{He} (a. u.)"}});
+    spectra.add("fProcessedEvents", "Nuclei - event filtered", HistType::kTH1F, {{4, -0.5, 3.5, "Event counter"}});
+  }
+
+  Filter collisionFilter = nabs(aod::collision::posZ) < cfgCutVertex;
+  Filter trackFilter = (nabs(aod::track::eta) < cfgCutEta) && (aod::track::isGlobalTrack == static_cast<uint8_t>(1u));
+
+  using TrackCandidates = soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::pidRespTPCDe, aod::pidRespTPCTr, aod::pidRespTPCHe, aod::pidRespTPCAl, aod::pidRespTOFDe, aod::pidRespTOFTr, aod::pidRespTOFHe, aod::pidRespTOFAl, aod::TrackSelection>>;
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels>>::iterator const& collision, aod::BCsWithTimestamps const&, TrackCandidates const& tracks)
+  {
+    // collision process loop
+    bool keepEvent[nNuclei]{false, false, false, false};
```

Comment:
`{false}` should auto-expand into a n element array if I am not mistaken

Reply:
Indeed, funny things (that burned me once) it is not true if you do it with `{true}`... we will fix this!

Reply:
Done